### PR TITLE
ENH: Support setting the backend per-test. (Fixes #35)

### LINKED ISSUE #35: Set Agg backend?
Does it make sense for pytest-mpl to set the backend to "Agg" in there somewhere? I don't see a usecase where it doesn't make sense, but that doesn't mean I'm not missing something. I just want to make my tests able to run headless a bit more easily.

### PR DESCRIPTION
**Pull request by dopplershift**
This works by using matplotlib's "experimental" support for switching
backends, and should only be used to switch between image backends.

### CODE REVIEW

File: pytest_mpl/plugin.py [hunk 1]
```diff
@@ -131,7 +144,7 @@ def item_function_wrapper(*args, **kwargs):
 
             baseline_remote = baseline_dir.startswith('http')
 
-            with plt.style.context(style):
+            with plt.style.context(style), switch_backend(backend):
```
Comment (astrofrog) [inline pytest_mpl/plugin.py]: @dopplershift looks like this is causing issues with Python 2.6. Does Matplotlib still support Python 2.6?

Comment (dopplershift) [inline pytest_mpl/plugin.py]: Only on the 1.4 series--so I guess the question is when does pytest-mpl drop 1.4?

Comment (astrofrog) [inline pytest_mpl/plugin.py]: Now 2.0 is out, I'd be fine with dropping 1.4 (I normally like keeping compatibility with the latest two versions)

**Comment by dopplershift** [comment]
I've gone ahead and removed all traces of Python 2.6 and Matplotlib 1.4--I'm hoping that's what you were looking for.

Rebased on latest master.

**Comment by dopplershift** [comment]
This isn't introducing any new test failures. Apparently, something's not quite right with testing against matplotlib 2.0.

**Comment by astrofrog** [comment]
@dopplershift - can you rebase on the latest master? (I've merged in changes that should fix Travis)

**Comment by dopplershift** [comment]
Linux tests are failing because conda can't find packages for numpy 1.9 and matplotlib 1.5. Thoughts?

**Comment by dopplershift** [comment]
Anything more you'd like to see here?

File: .travis.yml [hunk 2]
```diff
@@ -13,10 +13,8 @@ env:
     - CONDA_DEPENDENCIES="pytest matplotlib nose coverage"
     - PIP_DEPENDENCIES="pytest-cov coveralls"
   matrix:
-    - PYTHON_VERSION=2.6 MATPLOTLIB_VERSION=1.4 NUMPY_VERSION=1.9
-    - PYTHON_VERSION=2.7 MATPLOTLIB_VERSION=1.4 NUMPY_VERSION=1.9
-    - PYTHON_VERSION=3.3 MATPLOTLIB_VERSION=1.4 NUMPY_VERSION=1.9
-    - PYTHON_VERSION=3.4 MATPLOTLIB_VERSION=1.4 NUMPY_VERSION=1.9
+    - PYTHON_VERSION=3.3 MATPLOTLIB_VERSION=1.5 NUMPY_VERSION=1.9
+    - PYTHON_VERSION=3.4 MATPLOTLIB_VERSION=1.5 NUMPY_VERSION=1.9
```
Comment (astrofrog) [inline .travis.yml]: @dopplershift - try not pinnng the Numpy version here

**Comment by dopplershift** [comment]
No conda packages for Python 3.3 for matplotlib 1.5. Willing to drop Python 3.3? We'd still have the last 3 versions of the 3.x series.

**Comment by bsipocz** [comment]
@dopplershift - You can always move mpl from ``CONDA_DEPENDENCIES`` to ``PIP_DEPENDENCIES`` for the py3.3 run.

**Comment by dopplershift** [comment]
True, but probably best to make a firm decision on 3.3 support, or at least testing for it.

**Comment by astrofrog** [comment]
Does Matplotlib support 3.3?

**Comment by dopplershift** [comment]
As far back as 1.5, matplotlib is >= 3.4: http://matplotlib.org/1.5.3/users/installing.html

**Comment by astrofrog** [comment]
Ok then yes let's drop 3.3

**Comment by dopplershift** [comment]
All testing passing now.

**Comment by dopplershift** [comment]
Are we waiting on anything here?

**Comment by astrofrog** [comment]
(on vacation but will review/merge on my return!)

**Comment by jnothman** [comment]
Since #47, plt and matplotlib are only available locally, so there's a logical merge conflict in this PR.

Since matplotlib is only imported locally now, the user at least has the option to import matplotlib globally and set the backend once as a fixture, prior to it being imported by pytest-mpl.

I also wonder if we can more generally allow the user to specify arguments for `mpl.rc_context` here... But I assume `switch_backend` is necessary regardless of rc in order for an already imported pyplot to behave.

**Comment by astrofrog** [comment]
@dopplershift - this looks good, but as @jnothman pointed out this won't work as-is due to #47, so could you rebase and make sure that the tests pass locally?

**Comment by dopplershift** [comment]
Rebased and tests pass locally.

**Comment by dopplershift** [comment]
Looks like we're good here.

**Comment by astrofrog** [comment]
Thanks @dopplershift!

**Pull request merged**